test/service.cc

```diff
@@ -482,67 +378,6 @@ TEST_CASE("unified_sites placeholder") {
         */
     }
 }
-TEST_CASE("genotyper with ref changes") {
-    unique_ptr<VCFData> data;
-    Status s = VCFData::Open({"joint_A.gvcf", "joint_B.gvcf", "joint_C.gvcf", "joint_D.gvcf"}, data);
-    REQUIRE(s.ok());
-    unique_ptr<Service> svc;
-    s = Service::Start(*data, *data, svc);
-    REQUIRE(s.ok());
-
-    discovered_alleles als;
-
-    const string tfn("/tmp/GLnexus_unit_tests.bcf");
-
-    s = svc->discover_alleles("<ALL>", range(0, 0, 1000000), als);
-    REQUIRE(s.ok());
-
-    vector<unified_site> sites;
-    s = unified_sites(als, sites);
-    REQUIRE(s.ok());
-
-    consolidated_loss losses;
-    s = svc->genotype_sites(genotyper_config(), string("<ALL>"), sites, tfn, losses);
-    REQUIRE(s.ok());
-
-    unique_ptr<vcfFile, void(*)(vcfFile*)> vcf(bcf_open(tfn.c_str(), "r"), [](vcfFile* f) { bcf_close(f); });
-    REQUIRE(bool(vcf));
-
-    shared_ptr<bcf_hdr_t> hdr(bcf_hdr_read(vcf.get()), &bcf_hdr_destroy);
-    REQUIRE(bool(hdr));
-
-    int nsamples = bcf_hdr_nsamples(hdr);
-    REQUIRE(nsamples == 4);
-    REQUIRE(string(bcf_hdr_int2id(hdr.get(), BCF_DT_SAMPLE, 0)) == "JOINT_A");
-    REQUIRE(string(bcf_hdr_int2id(hdr.get(), BCF_DT_SAMPLE, 1)) == "JOINT_B");
-    REQUIRE(string(bcf_hdr_int2id(hdr.get(), BCF_DT_SAMPLE, 2)) == "JOINT_C");
-    REQUIRE(string(bcf_hdr_int2id(hdr.get(), BCF_DT_SAMPLE, 3)) == "JOINT_D");
-
-    unique_ptr<bcf1_t,void(*)(bcf1_t*)> record(bcf_init(), &bcf_destroy);
-    REQUIRE(bcf_read(vcf.get(), hdr.get(), record.get()) == 0);
-    REQUIRE(bcf_unpack(record.get(), BCF_UN_ALL) == 0);
-
-    REQUIRE(record->n_allele == 4);
-    REQUIRE(string(record->d.allele[0]) == "ATGTGTG");
-    REQUIRE(string(record->d.allele[1]) == "ATGTGTGTG");
-    REQUIRE(string(record->d.allele[2]) == "ATGTG");
-    REQUIRE(string(record->d.allele[3]) == "A");
-
-    int *gt = nullptr, gtsz = 0;
-    int nGT = bcf_get_genotypes(hdr.get(), record.get(), &gt, &gtsz);
-    REQUIRE(nGT == 8);
-
-    // Because the REF has changed in JOINT_A, both ALT alleles
-    // should be lost instead of being interpreted as is
-    REQUIRE(bcf_gt_allele(gt[0]) == 3); // 2 and 3 should be flipped here...
```

Comment:
Let us not forget the small detail alluded to in this comment: when we emit an unphased heterozygous genotype, by convention we should write the two alleles indices in increasing order, e.g. "2/3" instead of "3/2".

In general, the total order of alleles in the unified site may not be consistent with the partial order in each input gVCF, so we'll need to go out of our way to swap the order when necessary. 

We should construct some test that elicits this, but we can file that away as a small detail to handle later (obviously the placeholder genotyper did not do it, either)